api/src/info.rs

```diff
@@ -56,6 +56,12 @@ pub struct BootInfo {
     pub ramdisk_addr: Optional<u64>,
     /// Ramdisk image size, set to 0 if addr is None
     pub ramdisk_len: u64,
+    /// Kernel image address
+    pub kernel_addr: u64,
+    /// Kernel image size
+    pub kernel_len: u64,
+    /// Kernel image relocation address
+    pub kernel_image_offset: u64,
```

Comment:
These fields need more explicit documentation. Some important points that we need to mention are:

- The `kernel_addr` and `kernel_len` describe the location of the raw ELF file in memory, not the loaded kernel.
- `kernel_addr` is a physical address, not a virtual address such as most other fields in the boot info.
- The `kernel_image_offset` is an offset in the virtual address space. It affects the loaded kernel image, not the raw ELF file.